### PR TITLE
feat: borsh sized serialization should be fallible  

### PR DESCRIPTION
**Pull request by jorgeantonio21**
Description
---
Adds fallibility to borsh serialized size method. 

Motivation and Context
---
Audit

How Has This Been Tested?
---

What process can a PR reviewer use to test or verify this change?
---

Breaking Changes
---

- [x] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [ ] Other - Please specify

### CODE REVIEW

File: base_layer/core/src/transactions/aggregated_body.rs [hunk 1]
```diff
@@ -359,7 +359,10 @@ impl AggregateBody {
     }
 
     pub fn sum_features_and_scripts_size(&self) -> usize {
-        self.outputs.iter().map(|o| o.get_features_and_scripts_size()).sum()
+        self.outputs
+            .iter()
+            .map(|o| o.get_features_and_scripts_size().expect("Failed to serialize size"))
```
Comment (SWvheerden) [inline base_layer/core/src/transactions/aggregated_body.rs]: prefer making all of these errors.

File: base_layer/core/src/transactions/test_helpers.rs [hunk 2]
```diff
@@ -189,7 +189,12 @@ impl TestParams {
     pub fn get_size_for_default_features_and_scripts(&self, num_outputs: usize) -> usize {
         let output_features = OutputFeatures { ..Default::default() };
         self.fee().weighting().round_up_features_and_scripts_size(
-            script![Nop].get_serialized_size() + output_features.get_serialized_size(),
+            script![Nop]
+                .get_serialized_size()
+                .expect("Failed to get serialized size") +
+                output_features
+                    .get_serialized_size()
+                    .expect("Failed to get serialized size"),
```
Comment (SWvheerden) [inline base_layer/core/src/transactions/test_helpers.rs]: same here

File: base_layer/core/src/transactions/test_helpers.rs [hunk 3]
```diff
@@ -528,9 +533,15 @@ pub async fn create_wallet_outputs(
...
-            output_covenant.get_serialized_size(),
+        output_features
+            .get_serialized_size()
+            .expect("Failed to get serialized size") +
+            output_script
+                .get_serialized_size()
+                .expect("Failed to get serialized size") +
+            output_covenant
+                .get_serialized_size()
+                .expect("Failed to get serialized size"),
```
Comment (SWvheerden) [inline base_layer/core/src/transactions/test_helpers.rs]: same here

**Review by SWvheerden — COMMENTED** [review]
This is going the correct way. 
But we should remove the expects as they are the same as the unwrap we replace. 
I would prefer to replace them with Result<>

File: base_layer/core/src/transactions/aggregated_body.rs (hunk 1 above)
Comment (jorgeantonio21) [reply base_layer/core/src/transactions/aggregated_body.rs]: Refactored :) !

**Review by SWvheerden — DISMISSED** [review]
It is so nice to see all these unwraps disappear

**Pull request merged**